wai-extra/Network/Wai/Middleware/RequestLogger.hs

```diff
@@ -234,6 +238,14 @@ logStdout = unsafePerformIO $ mkRequestLogger def { outputFormat = Apache FromSo
 logStdoutDev :: Middleware
 logStdoutDev = unsafePerformIO $ mkRequestLogger def
 
+-- | Development request logger middleware.
+--
+-- This uses the 'Detailed' 'True' logging format and logs to 'stdout'.
+-- @since 3.1.7
+{-# NOINLINE logStdoutDevWithRequestPrelogging #-}
+logStdoutDevWithRequestPrelogging :: Middleware
```

Comment:
I'm not sure it's worth adding in a new value here, versus letting people use `mkRequestLogger` directly for this case.

Reply:
Fair enough! I'll remove it. I guess I'd like it to be very convenient to handle this scenario, but on second thought this is mainly due to the situation where logging is cancelled due to an exception.

Reply:
I should point out: the exception scenario is almost certainly a mistake. I know at least Servant mistakenly does not handle exceptions and propagates them up to Warp for handling, which is wrong. You may be encountering that issue here. If so, I'd recommend wrapping the application in an exception handler. Warp is ill equipped to handle exceptions, and middleware ideally shouldn't have to handle those cases at all.

Reply:
Perhaps the best solution would be a new exception handling middleware?

Reply:
I'm overall -1 on this, since it really should be handled in the `Application` itself. The code in question here doesn't handle some painful corner cases such as "what if the exception handler throws an exception?"